R/dash.R

```diff
@@ -713,14 +715,24 @@ Dash <- R6::R6Class(
       if (hot_reload == TRUE & !(is.null(source_dir))) {
         self$server$on('cycle-end', function(server, ...) {
           # handle case where assets are not present, since we can still hot reload the app itself
-          # private$last_refresh will get set after the asset_map is refreshed
+          #
+          # private$last_refresh is set after the asset_map is refreshed
+          # private$last_reload stores the time of the last hard or soft reload event
           # private$last_cycle will be set when the cycle-end handler terminates
-          if (!is.null(private$last_cycle) & !is.null(hot_reload_interval)) {
-            # determine if the time since last cycle end is equal to or longer than the requested check interval
-            permit_reload <- (as.integer(Sys.time()) - private$last_cycle) >= hot_reload_interval
+          #
+          if (!is.null(private$last_cycle) & !is.null(hot_reload_watch_interval)) {
+            if (!private$has_reloaded)
+              # allow reloading on first change, then compare against the hot_reload_watch_interval
+              # to control whether future reloads are permitted on cycle end
+              permit_reload <- TRUE
+            else {
+              # determine if the time since last reload end is equal to or longer than the requested check interval
+              # (Sys.time() - private$last_reload) provides time in seconds, same units as hot_reload_watch_interval
+              permit_reload <- (Sys.time() - private$last_reload) >= hot_reload_watch_interval
```

Comment:
This will work. But a common way to do it with only one variable is to start with `private$last_reload = 0` or some known distant past time (0 often works for that purpose because it typically means the unix epoch = 1/1/1970)

Reply:
fixed in 0ce7a6d